### PR TITLE
Add a Shared Wait Signal

### PR DESCRIPTION
**Pull request by pasitlaothamatas**
### Summary

- Added a shared wait signal to deal with situations like DCM-12
- Add unit tests to check that the buzzer is turned on for 2 seconds after entering the drive state to check that the wait signal is operating correctly

### Changelist 

- Added a flag for the buzzer struct to indicate when the buzzer has been turned on/off

### Testing Done

 - Create unit tests for the wait signal in `Test_SharedWaitSignal.cpp`
- Turn the buzzer on in `DriveStateRunOnEntry` in `App_DriveState`
- Test that the wait signal is working properly by checking that the buzzer is turned on for two seconds after entering the drive state

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [ ] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [ ] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

### CODE REVIEW

File: boards/DCM/Src/App/App_DcmWorld.c [hunk 1]
```diff
@@ -13,6 +14,7 @@ struct DcmWorld
     struct BrakeLight *       brake_light;
     struct Buzzer *           buzzer;
     struct ErrorTable *       error_table;
+    struct WaitSignal *       buzzer_wait_signal;
```
Comment (pasitlaothamatas) [inline boards/DCM/Src/App/App_DcmWorld.c:17]: will add a LL library if required. This should suffice for now

File: boards/shared/Test/Src/Test_SharedWaitSignal.cpp [hunk 2]
```diff
@@ -0,0 +1,250 @@
+#include "Test_Shared.h"
+
+extern "C"
+{
+#include "App_SharedWaitSignal.h"
+}
+
+// TODO: Refactor fake function names after 668 is merged
```
Comment (markomarkusi) [inline boards/shared/Test/Src/Test_SharedWaitSignal.cpp:8]: :eyes: :eyes: :eyes: :eyes: :eyes: :eyes:

File: boards/DCM/Inc/App/App_DcmWorld.h [hunk 3]
```diff
@@ -28,7 +30,10 @@ struct DcmWorld *App_DcmWorld_Create(
     struct BrakeLight *       brake_light,
     struct Buzzer *           buzzer,
     struct ErrorTable *       error_table,
-    struct Clock *            clock);
+    struct Clock *            clock,
+
+    bool (*is_buzzer_on)(struct DcmWorld *),
+    void (*buzzer_callback)(struct DcmWorld *));
```
Comment (markomarkusi) [inline boards/DCM/Inc/App/App_DcmWorld.h:36]: LGTM did you just want to change the order here as a NIT?

**Review by markomarkusi — APPROVED** [review]

File: boards/shared/Test/Src/Test_SharedWaitSignal.cpp (hunk 2 above)
Comment (pasitlaothamatas) [reply boards/shared/Test/Src/Test_SharedWaitSignal.cpp:8]: oh whoops forgot to delete this

File: boards/DCM/Inc/App/App_DcmWorld.h (hunk 3 above)
Comment (pasitlaothamatas) [reply boards/DCM/Inc/App/App_DcmWorld.h:36]: Do we currently have a convention for this?

Comment (markomarkusi) [reply boards/DCM/Inc/App/App_DcmWorld.h:36]: No i don't think so it just looks nicer to group the buzzer stuff together than have part of it after the clock structure

File: boards/DCM/Src/App/App_DcmWorld.c [hunk 4]
```diff
@@ -24,9 +26,12 @@ struct DcmWorld *App_DcmWorld_Create(
...
     struct Buzzer *const            buzzer,
     struct ErrorTable *const        error_table,
-    struct Clock *const             clock)
+    struct Clock *const             clock,
+
+    bool (*is_buzzer_on)(struct DcmWorld *),
+    void (*buzzer_complete_callback)(struct DcmWorld *))
 {
-    struct DcmWorld *world = (struct DcmWorld *)malloc(sizeof(struct DcmWorld));
+    struct DcmWorld *world = malloc(sizeof(struct DcmWorld));
```
Comment (pasitlaothamatas) [inline boards/DCM/Src/App/App_DcmWorld.c:34]: Got rid of casting here https://stackoverflow.com/questions/605845/do-i-cast-the-result-of-malloc

File: boards/DCM/Inc/App/App_DcmWorld.h (hunk 3 above)
Comment (pasitlaothamatas) [reply boards/DCM/Inc/App/App_DcmWorld.h:36]: I prefer this cuz it splits the members in world from the callback related signals

**Comment by pasitlaothamatas** [comment]
This was approved but should prob review a couple of things before I merge. Ready for another round.

File: boards/shared/Test/Src/Test_SharedWaitSignal.cpp (hunk 2 above)
Comment (pasitlaothamatas) [reply boards/shared/Test/Src/Test_SharedWaitSignal.cpp]: This should be deleted

**Pull request merged**